include/fmt/ranges.h

```diff
@@ -0,0 +1,224 @@
+// Formatting library for C++ - the core API
+//
+// Copyright (c) 2012 - present, Victor Zverovich
+// All rights reserved.
+//
+// For the license information refer to format.h.
+//
+// Copyright (c) 2018 - present, Remotion (Igor Schulz)
+// All Rights Reserved
+// {fmt} support for ranges, containers and types tuple interface.
+
+#ifndef FMT_RANGES_H_
+#define FMT_RANGES_H_
+
+#include "format.h"
+#include <type_traits>
+
+// output only up to N items from the range.
+#ifndef FMT_RANGE_OUTPUT_LENGTH_LIMIT
+#define FMT_RANGE_OUTPUT_LENGTH_LIMIT 256
+#endif
+
+namespace fmt {
+
+template <typename Char>
+struct formatting_base {
+	template <typename ParseContext>
+	FMT_CONSTEXPR auto parse(ParseContext &ctx) -> decltype(ctx.begin()) { return ctx.begin(); }
+};
+
+template <typename Char, typename Enable = void>
+struct formatting_range : formatting_base<Char>
+{
+	Char prefix = '{';
+	Char delimiter = ',';
+	Char postfix = '}';
+	bool add_spaces = true;
+};
+
+template <typename Char, typename Enable = void>
+struct formatting_tuple : formatting_base<Char>
+{
+	Char prefix = '[';
+	Char delimiter = ',';
+	Char postfix = ']';
+	bool add_spaces = true;
+};
+
+template<typename RangeT, typename OutputIterator>
+void copy(const RangeT &range, OutputIterator out) {
+	for (const auto& it : range) {
+		*out++ = it;
+	}
+}
+
+template<typename OutputIterator>
+void copy(const char *str, OutputIterator out) {
+	const char* p_curr = str;
+	while (*p_curr) {
+		*out++ = *p_curr++;
+	}
+}
+
+template<typename OutputIterator>
+void copy(const char ch, OutputIterator out) {
+	*out++ = ch;
+}
+
+} // namespace fmt
+
+
+namespace fmt {
+namespace meta {
+
+/// Return true value if T has std::string interface, like std::string_view.
+template<typename T>
+class is_like_std_string {
+	template<typename U> static auto check(U* p) -> decltype(
+		p->find('a')
+		, p->length()
+		, p->data()
+		, int());
+	template<typename  > static void check(...);
+public:
+	static const bool value = !std::is_void< decltype(check<T>(nullptr)) >::value;
+};
+
+template<typename T>
+constexpr bool is_like_std_string_v = is_like_std_string<T>::value;
```

Comment:
Here and elsewhere: constexpr -> FMT_CONSTEXPR

Reply:
Done